mathics/core/assignment.py

```diff
@@ -159,9 +159,9 @@ def repl_pattern_by_symbol(expr):
 
     changed = False
     new_elements = []
-    for element in elements:
-        element = repl_pattern_by_symbol(element)
-        if not (element is element):
+    for _element in elements:
+        element = repl_pattern_by_symbol(_element)
+        if element != _element:
```

Comment:
Just a detail: to avoid some overhead in comparisons, we use the idiom
```
if not (element is _element):
```
instead of checking the inequality.

Reply:
Actually then it should be element is not _element then. I will put in a PR for this.